frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx

```diff
@@ -224,12 +273,16 @@ export const OperatorHubTileView: React.FC<OperatorHubTileViewProps> = (props) =
   const [detailsItem, setDetailsItem] = React.useState(null);
   const [showDetails, setShowDetails] = React.useState(false);
 
+  const itemsByArchAndOS = !canFilterByArchAndOS 
```

Comment:
I'd flip the logic here so you don't have to think about the `!`

```ts
const filteredItems = canFilterByArchAndOS 
  ? filterByArchAndOS(props.items)
  : props.items;
```

But I think it's probably cleaner to put this as an early return in the `filterByArchAndOS` function.

```ts
const filterByArchAndOS = (items: OperatorHubItem[]): OperatorHubItem[] => {
  if (!window.SERVER_FLAGS.GOARCH || !window.SERVER_FLAGS.GOOS) {
    return items;
  }
  // ...
```

Reply:
Will do! Agree, mine initial is leaky.